### PR TITLE
Approved list editorextensions

### PR DESCRIPTION
**Pull request by pelikhan**
This PR removes all the permission UI/logic for editor extensions, and replaces it with an approved list in targetconfig.json.

- [x] remove all permissions UI, messages, logic
- [x] enforce URL from targetconfig.json
- [x] updated docs

image
🥇

### CODE REVIEW

File: webapp/src/extensionManager.ts [hunk 1]
```diff
@@ -317,27 +124,21 @@ export async function resolveExtensionUrl(pkg: pxt.Package) {
...
     else if (extension.url
-        && packagesConfig?.approvedEditorExtensionUrls
+        && repoStatus === pxt.github.GitRepoStatus.Approved
         && packagesConfig?.approvedEditorExtensionUrls?.indexOf(extension.url) > -1) {
-        // custom url, but not support in editor
         pxt.log(`extension url ${extension.url} trusted`)
         url = extension.url;
-        // must also be approved to be trusted
-        trusted = repoStatus === pxt.github.GitRepoStatus.Approved;
-    } else if (!parsedRepo) {
```
Comment (pelikhan) [inline webapp/src/extensionManager.ts:330]: this is the code that allowed any github repo to be loaded

File: webapp/src/extensions.tsx [hunk 2]
```diff
@@ -113,22 +105,29 @@ export class Extensions extends data.Component<ISettingsProps, ExtensionsState>
...
-            this.send(extension, { target: pxt.appTarget.id, type: "pxtpkgext", event: "extshown" } as pxt.editor.ShownEvent);
-        })
-    }
+    showExtensionAsync(extension: string, url: string) {
+        // sanity check: URL must be approved in targetconfig.json
+        return pxt.targetConfigAsync()
+            .then(config => {
+
+                // sanity check: this should have been caught earlier
+                const packagesConfig = config?.packages;
```
Comment (pelikhan) [inline webapp/src/extensions.tsx:114]: final stop gap: make sure the URL is allowed

**Review by riknoll — APPROVED** [review]

**Comment by pelikhan** [comment]
@Brahma Giri Abhijith Chatra<mailto:abchatra@microsoft.com>  this can wait after arcade but wouldn’t hurt either.

________________________________

**Comment by abchatra** [comment]
Thanks! Lets wait this till arcade release (bit too many changes at the end will be riskly).

**Review by jwunderl — APPROVED** [review]
(yours is way better because it's all code but I still like mine from the other day, even if it was just built translations
image)

**Comment by pelikhan** [comment]
waiting on arcade release

**Pull request merged**